util/setup_gcp_project.sh

```diff
@@ -53,7 +51,4 @@ gcloud projects add-iam-policy-binding $PROJECT --member="serviceAccount:$CLOUDO
 gcloud projects add-iam-policy-binding $PROJECT --member="serviceAccount:$CLOUDOPS@$PROJECT.iam.gserviceaccount.com" --role='roles/monitoring.metricWriter'
 gcloud projects add-iam-policy-binding $PROJECT --member="serviceAccount:$CLOUDOPS@$PROJECT.iam.gserviceaccount.com" --role='roles/stackdriver.resourceMetadata.writer'
 gcloud projects add-iam-policy-binding $PROJECT --member="serviceAccount:$CLOUDOPS@$PROJECT.iam.gserviceaccount.com" --role='roles/monitoring.dashboardEditor'
-gcloud projects add-iam-policy-binding $PROJECT --member="serviceAccount:$SUPERADMIN@$PROJECT.iam.gserviceaccount.com" --role='roles/iam.serviceAccountAdmin'
-gcloud projects add-iam-policy-binding $PROJECT --member="serviceAccount:$SUPERADMIN@$PROJECT.iam.gserviceaccount.com" --role='roles/iam.serviceAccountKeyAdmin'
-gcloud projects add-iam-policy-binding $PROJECT --member="serviceAccount:$SUPERADMIN@$PROJECT.iam.gserviceaccount.com" --role='roles/resourcemanager.projectIamAdmin'
-gcloud projects add-iam-policy-binding $PROJECT --member="serviceAccount:$SUPERADMIN@$PROJECT.iam.gserviceaccount.com" --role='roles/editor'
+gcloud projects add-iam-policy-binding $PROJECT --member="serviceAccount:$BMCTL@$PROJECT.iam.gserviceaccount.com" --role='roles/compute.viewer'
```

Comment:
Are you sure this works? It didn't with the initial GA release. We had to have each of those roles for the account that was running `bmctl`

Reply:
I didn't test any of the setip_gcp_project.sh changes, I used the TF only approach. Curious about this.

Reply:
I tested and the cluster comes up for me with the reduced permissions on pre-provisioned keys (and the same with TF SA management, which also uses the reduced permissions). Of course it would be appreciated if you could also confirm 😃 . I suppose it's possible that a future addition to the permissions required by `bmctl` could cause this to break, but we version the release used, so the impact of any such future change would be minimal. It's also possible that additional permissions are required for operations other than cluster build (that this codebase doesn't currently make use of). I will look into this as an addition to the [docs](https://cloud.google.com/anthos/gke/docs/bare-metal/1.6/installing/creating-clusters/admin-cluster-creation#logging_into_gcloud_and_creating_a_admin_cluster_config_file) for when the automatic API enablement and Service Account creation features are not done by `bmctl` (as in our case here).